src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -310,7 +312,7 @@ CreateInsertSelectRouterPlan(Query *originalQuery,
 		modifyTask = RouterModifyTaskForShardInterval(originalQuery, targetShardInterval,
 													  relationRestrictionContext,
 													  taskIdIndex,
-													  restrictionEquivalenceForPartitionKeys);
+													  safeToPushDownSubquery);
```

Comment:
we are passing  variable `safeToPushDownSubquery ` in place of `allRelationsJoinedOnPartitionKey` perhaps we should change function api here